doc/pull_request_engagement_rule.md

```diff
@@ -0,0 +1,44 @@
+# Pull Request Engagement Rules
+
+## Why do we need these rules
+
+We have 20 PR (pull request) opened on an average day, more when it is closer to final days of a sprint. Reviewing pull request cost time and energy. Automation jobs triggered by the PR also constantly compete for the finite resources.
+
+To better facilitate the reviewing process to ensure PR are reviewed and closed in timely fashion the engineering team set up the engagement rules to define when and how we will begin review pull requests.
```

Comment:
`To better facilitate the reviewing process to ensure PRs are reviewed and closed in a timely fashion, the engineering team has set up the following engagement rules to define when and how we will begin reviewing pull requests.`